### PR TITLE
Index Pods by UID and join with containers based on this.

### LINKED ISSUE #822: Missing pod names in kubernetes' pod view
screen shot 2016-01-15 at 3 46 36 pm

### LINKED ISSUE #1395: Pause containers don't show as children of pods
the docker label "io.kubernetes.pod.uid" seems to tell us what pod they belong to.

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #1395, fixes #822

### CODE REVIEW

File: probe/kubernetes/pod.go [hunk 1]
```diff
@@ -88,14 +87,13 @@ func (p *pod) NodeName() string {
...
-	n := report.MakeNodeWith(report.MakePodNodeID(p.Namespace(), p.Name()), map[string]string{
-		PodID:           p.ID(),
-		PodName:         p.Name(),
-		Namespace:       p.Namespace(),
-		PodCreated:      p.Created(),
-		PodContainerIDs: strings.Join(p.ContainerIDs(), " "),
-		PodState:        p.State(),
-		PodIP:           p.Status.PodIP,
+	n := report.MakeNodeWith(report.MakePodNodeID(p.UID()), map[string]string{
+		PodID:      p.ID(),
```
Comment (paulbellamy) [inline probe/kubernetes/pod.go]: Do we use this anymore? Maybe we could remove it altogether...

File: probe/kubernetes/reporter_test.go [hunk 2]
```diff
@@ -44,6 +45,7 @@ var (
 		TypeMeta: podTypeMeta,
 		ObjectMeta: api.ObjectMeta{
 			Name:              "pong-b",
+			UID:               "pong-b",
```
Comment (paulbellamy) [inline probe/kubernetes/reporter_test.go]: these don't look like a uid

File: probe/kubernetes/reporter_test.go [hunk 3]
```diff
@@ -127,7 +129,8 @@ func (*mockClient) WalkNodes(f func(*api.Node) error) error {
 }
 func (*mockClient) WatchPods(func(kubernetes.Event, kubernetes.Pod)) {}
 func (c *mockClient) GetLogs(namespaceID, podName string) (io.ReadCloser, error) {
-	r, ok := c.logs[report.MakePodNodeID(namespaceID, podName)]
+	fmt.Println("here", namespaceID, podName)
```
Comment (paulbellamy) [inline probe/kubernetes/reporter_test.go]: forgot debug

File: probe/kubernetes/reporter_test.go [hunk 4]
```diff
@@ -219,34 +220,6 @@ func TestReporter(t *testing.T) {
...
-			// container should have namespace
-			if have, ok := node.Latest.Lookup(kubernetes.Namespace); !ok || have != "ping" {
-				t.Errorf("Expected container %s latest %q: %q, got %q", containerID, kubernetes.Namespace, "ping", have)
-			}
-			// container should have pod parent
-			if parents, ok := node.Parents.Lookup(report.Pod); !ok || !parents.Contains(pod.nodeID) {
-				t.Errorf("Expected container %s to have parent service %q, got %q", containerID, pod.nodeID, parents)
-			}
-		}
-	}
```
Comment (paulbellamy) [inline probe/kubernetes/reporter_test.go]: Maybe move these into a test for `Reporter.Tag`...

File: probe/kubernetes/pod.go (hunk 1 above)
Comment (tomwilkie) [inline probe/kubernetes/pod.go]: Still used to construct the details node (label I think)

**Comment by tomwilkie** [comment]
@paulbellamy PTAL

**Comment by paulbellamy** [comment]
LGTM

**Pull request merged**